### PR TITLE
lnwire+peer: fully validate node aliases on the wire, don't d/c due to invalid aliases

### PR DESCRIPTION
**Pull request by Roasbeef**

### CODE REVIEW

File: peer.go [hunk 1]
```diff
@@ -996,6 +996,12 @@ out:
 				idleTimer.Reset(idleTimeout)
 				continue
 
+			// If the NodeAnnouncement has an invalid alias, then
+			// we'll log that error above and continue so we can
+			// continue to read messges from the peer.
+			case lnwire.ErrInvalidNodeAlias:
```
Comment (halseth) [inline peer.go]: should `idleTimer` be reset, as in the other cases?

Comment (Roasbeef) [reply peer.go]: Fixed!

**Comment by Roasbeef** [comment]
Comment addressed, PTAL @halseth

File: peer.go [hunk 2]
```diff
@@ -996,6 +996,13 @@ out:
...
```
Comment (halseth) [inline peer.go]: This should have a test if possible. I think maybe this should be a pointer as well?

Comment (Roasbeef) [reply peer.go]: There isn't a good way to make an accurate end to end test for this as we don't allow local nodes to set the value, and we don't yet have a p2p testing framework in place. Correct on pointer, just added! We do currently have unit tests that cover validation however.

**Pull request merged**